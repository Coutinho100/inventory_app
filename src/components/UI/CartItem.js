import React from 'react';
import {
    View,
    Text,
    StyleSheet,
    TouchableOpacity,
} from 'react-native';
import Card from '../Component/Card'
import { Entypo, FontAwesome5 } from '@expo/vector-icons';
const CartItem = props => {
    return (

        <View>
            <Card style={styles.cartItem}>
                <View style={styles.itemData}>
                    <FontAwesome5 name="capsules" size={24} color="black" />
                </View>
                <View>
                    <Text style={styles.mainText}>{props.title}</Text>
                    <Text style={styles.quantity}>Qty:{props.quantity} </Text>
                </View>
                <View style={styles.itemData}>
                    {/* <Text style={styles.mainText}>${props.amount.toFixed(2)}</Text> */}
                    {props.deletable && (
                        <TouchableOpacity
                            onPress={props.onRemove}
                            style={styles.deleteButton}
                        >
                            <Entypo name="cross" size={24} color="black" size={23}
                                color="red" />
                        </TouchableOpacity>
                    )}
                </View>
            </Card>
        </View>

    );
};

const styles = StyleSheet.create({
    cartItem: {
        padding: 10,
        backgroundColor: 'white',
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginHorizontal: 20,
        marginBottom: 20
    },
    itemData: {
        flexDirection: 'row',
        alignItems: 'center'
    },
    quantity: {
        fontFamily: 'open-sans',
        color: '#888',
        fontSize: 16,
        paddingLeft: 10
    },
    mainText: {
        fontFamily: 'open-sans-bold',
        fontSize: 16
    },
    deleteButton: {
        marginLeft: 20
    }
});

export default CartItem;
