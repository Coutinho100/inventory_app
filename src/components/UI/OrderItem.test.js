import React from 'react';
import OrderItem from './OrderItem';
import {
    configure,
    shallow
} from 'enzyme';
import { Text, View, FlatList } from 'react-native';
import EnzymeAdapter from 'enzyme-adapter-react-16';
import test_function, {findByTestAttr} from '../../_test_/componenets/test_function'
import Card from '../Component/Card';
import CartItem from './CartItem';

// JestHook.mock('expo-font');

configure({
    adapter: new EnzymeAdapter
})
const function_click=jest.fn()



    
const dummy_data = {
listData:["a","b","c"],
items:[{
    image:"image3.png",
    quantity:43,
    itemTitle:"hello3",
    price:343,
    company:"abc3",
    itemId:"id1"
},
{
    image:"image2.png",
    quantity:42,
    itemTitle:"hello2",
    price:342,
    company:"abc2",
    itemId:"id2"
},
{
    image:"image1.png",
    quantity:41,
    itemTitle:"hello1",
    price:341,
    company:"abc1",
    itemId:"id3"
},],
catTitle:"titles",
date:"Nov 3, 2020",
navigation:{navigate:function_click}
}

let extra=
describe('<OrderItem />', () => {
  it("Props and children in OrderItem", () =>
        {
            const wrapper = shallow(<OrderItem  {...dummy_data} />).dive()

            let cartItem= wrapper.find(CartItem)
            expect(cartItem.at(0).prop("quantity")).toEqual(43)
            expect(cartItem.at(2).prop("title")).toEqual("hello1")
            expect(cartItem.at(1).prop("quantity")).toEqual(42)
            expect(cartItem).toHaveLength(3)

        
            });
        });

test_function(["<OrderItem/>",2,1,1],['renders correctly','renders the length of view'],[<OrderItem items={[]}/>,View,Card,Text],extra)

