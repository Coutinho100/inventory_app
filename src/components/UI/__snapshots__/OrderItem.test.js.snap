// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<OrderItem/> renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "white",
      "borderRadius": 10,
      "elevation": 5,
      "margin": 20,
      "padding": 10,
      "shadowColor": "black",
      "shadowOffset": Object {
        "height": 2,
        "width": 0,
      },
      "shadowOpacity": 0.26,
      "shadowRadius": 8,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
        "marginBottom": 15,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#888",
          "fontFamily": "open-sans",
          "fontSize": 16,
        }
      }
    />
  </View>
  <View
    style={
      Object {
        "width": "100%",
      }
    }
  />
</View>
`;
